### DISCUSSION TITLE
Marine forecast swell and wave data in comparison to surf forecast apps

### DISCUSSION THREAD

**Discussion by aekong** [open-meteo/open-meteo#577] [Q&A]
Hello! 

First, thank you for this very awesome api service!

I had a quick question about wave and swell height forecast data.

I aggregated the 7 day marine forecast data into something more readable for myself.  The wave and swell forecast doesn't seem to line up with data from other apps that I've checked. For example, for Friday 1/5/24, for lat/lng 37.75/-122.5, my aggregate data looks like this:
```
{
    "time": "2024-01-05",
    "hourly": [
        {
            "time": "2024-01-05T00:00",
            "wave_height": 3.806,
            "wave_period": 12.45,
            "swell_wave_height": 3.806,
            "swell_wave_period": 12.45
        },
        ...
    ]
}
```
but i'm seeing different swell height and period data from other surf forecast sites for 1/5 midnight:

- surfline.com: 7.1ft, 13s
- surf-forecast.com: 6ft, 13s
- willyweather.com: 6.6ft, 13s
- windy.app: 6.8ft, 9s

The period is close enough, but the wave and swell height seem to have a pretty sizeable discrepancy.  I'm pretty new at reading surf data, though, so please forgive my ignorance if I am way off base. I understand that forecasts will differ by app/site, but am unclear on what I'm seeing in comparison.

Thanks!

**Comment by patrick-zippenfenig** [comment]
Hi!

I get around 7ft:

![Screenshot 2024-01-03 at 12 09 27](https://github.com/open-meteo/open-meteo/assets/11670456/1cfd9e0e-ba0d-45e9-bbd2-7d41703bf036)

Could you double check your coordinates and units?

**Reply by aekong** [reply]
Ah yes, that's for Jan 3.  The date I was looking at was Jan 5.  Came back to edit to add some screenshots and to say thank you for the very quick response!

<img width="1464" alt="Screenshot 2024-01-03 at 3 56 02 PM" src="https://github.com/open-meteo/open-meteo/assets/19291154/63e444ec-52e6-40c1-ae09-ac300af6cf5b">

For comparison, here are some other graphs:

surfline.com
<img width="1635" alt="Screenshot 2024-01-03 at 4 49 18 PM" src="https://github.com/open-meteo/open-meteo/assets/19291154/4f828a56-b12a-4c5c-85c3-96559a87d0d2">

willyweather.com:
<img width="737" alt="Screenshot 2024-01-03 at 3 58 42 PM" src="https://github.com/open-meteo/open-meteo/assets/19291154/87cb70d8-aa27-4ec8-930d-1175fc4c04b9">

surf-forecast.com
<img width="307" alt="Screenshot 2024-01-03 at 4 48 20 PM" src="https://github.com/open-meteo/open-meteo/assets/19291154/9882b1a4-9278-4389-aef6-987e6aec9163">

**Reply by aekong** [reply]
i do see from surf-forecast.com that there is data for 3 swells, each accounting for a different wind direction (i think?) and they vary quite a bit from each other.  maybe that has something to do with it? if we total those three waves and average it out, we'd get closer to the value from the open-meteo response, but i don't know if that's a valid calculation (again, please forgive my ignorance 😅)

**Reply by bastienlabelle** [reply]
Hi @aekong 

I'm the lead developer of a surf forecasting website in France, and since this question has never really been answered, I thought I'd drop a bit of knowledge on why data might be different on some surfing forecasts websites.

There are indeed 4 swells included in the GFS Wave data:

* Primary, secondary, and tertiary swells: generated far offshore by heavy storms.
* The wind swell partition: generated closer to the coast.

From my experience, most surf forecasting sites use GFS Wave data at a global scale. The data is typically rounded to the nearest foot for easier readability.

Except for Surfline, which uses its own proprietary model (LOTUS). Their model — when possible — takes into account the interaction between multiple swells affecting the primary swell, as well as the impact of bathymetry (seafloor topography), something GFS Wave does not really handle.

Also, the wave_height and wave_period values are averages calculated from all swell components:
It’s possible (and not uncommon) to have a situation where GFS Wave shows 6ft/13s, but in reality, there’s a primary swell at 6ft/9s and a secondary swell at 3ft/13s.
So in terms of actual surf conditions, it won't really feel like a clean 6ft/13s.

So this is why the data in most surfing app will look very similar to the data from GFSWave in Open-Meteo, except for Surfline who differs a bit. And none of them are of course 100% reliable ;)

**Reply by patrick-zippenfenig** [reply]
Note: Secondary swell was just added 2 weeks ago to the Open-Meteo API. Currently, it is only integrated for MeteoFrance wave models. https://github.com/open-meteo/open-meteo/pull/1309. For completeness I just added GFS wave secondary and tertiary swell components. Data should be available in the next hours: https://github.com/open-meteo/open-meteo/pull/1323
